blackboxopt/optimizers/botorch_base.py

```diff
@@ -277,15 +280,20 @@ def generate_evaluation_specification(self) -> EvaluationSpecification:
 
         if self.num_initial_random > 0 and (
             self.X.size(-2) < self.num_initial_random
-            or torch.nonzero(~torch.any(self.losses.isnan(), dim=1)).numel() == 0
+            or (
+                torch.nonzero(~torch.any(self.losses.isnan(), dim=1)).numel()
+                < self.num_initial_random
+            )
```

Comment:
Note, that the comment below now is wrong, because sampling random configurations now continues until `num_initial_random` many have all valid objectives.